test/server/cards/01_SOR/ImperialInterceptor.spec.js

```diff
@@ -0,0 +1,60 @@
+describe('Imperial Interceptor', function() {
+    integration(function() {
+        describe('Imperial Interceptor\'s When Played ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['imperial-interceptor'],
+                        groundArena: ['pyke-sentinel'],
+                        spaceArena: ['system-patrol-craft']
+                    },
+                    player2: {
+                        groundArena: ['wampa', 'superlaser-technician'],
+                        spaceArena: ['gladiator-star-destroyer']
+                    }
+                });
+            });
+
+            it('can only select space units & can select itself', function () {
```

Comment:
Because our tests require a lot of setup (each test case has to start an entire game instance), we try to consolidate test cases down as much as we can. This is not normal practice since typically you want test cases to be small, but since our test framework needs some improvement we need to handle things a little differently.

In addition, we typically assume that if a unit is selectable at the target phase that the effect would work correctly for any selectable card. Obviously there are exceptions if some card is very different than the others - in this case it may be worth having a separate test which damages the interceptor to confirm that the timing is working correctly.

To summarize, these are the test cases I would recommend for this card:
1. A test that resolves the ability, where we have one `toBeAbleToSelectExactly` check, check that the 'Pass ability' button is enabled, and then choose a unit which is not the interceptor and check that the damage is applied
2. Same test as 1 (we could skip the `toBeAbleToSelectExactly`) but we select the interceptor for the damage and check that it is in discard at the end
3. A test where we pass the ability (you already have this one)

Reply:
test updated